File: 1701-1800/1738-Kth-Largest-Value/1738.java
class Solution {
  public int kthLargestValue(int[][] matrix, int k) {
    int m = matrix.length, n = matrix[0].length;
    int[][] pre = new int[m + 1][n + 1];
    List<Integer> results = new ArrayList<Integer>();
    for (int i = 1; i <= m; ++i) {
      for (int j = 1; j <= n; ++j) {
        pre[i][j] = pre[i - 1][j] ^ pre[i][j - 1] ^ pre[i - 1][j - 1] ^ matrix[i - 1][j - 1];
        results.add(pre[i][j]);
      }
    }

    Collections.sort(results, new Comparator<Integer>() {
      public int compare(Integer num1, Integer num2) {
        return num2 - num1;
      }
    });
    return results.get(k - 1);
  }
}

File: 1701-1800/1738-Kth-Largest-Value/1738.go
func kthLargestValue(matrix [][]int, k int) int {
	m, n := len(matrix), len(matrix[0])
	results := make([]int, 0, m * n)
	pre := make([][]int, m + 1)
	pre[0] := make([]int, n + 1)
	for i, row := range matrix {
		pre[i + 1] = make([]int, n + 1)
		for j, val := range row {
			pre[i+1][j+1] = pre[i+1][j] ^ pre[i][j+1] ^ pre[i][j] ^ val
			results = append(results, pre[i+1][j+1])
		}
	}
	sort.Sort(sort.Reverse(sort.IntSlice(results)))
	return results[k-1]
}